codalab/worker/local_run/docker_image_manager.py

```diff
@@ -136,6 +136,32 @@ def get(self, image_spec):
         :param image_spec: Repo image_spec of docker image being requested
         :returns: A DockerAvailabilityState object with the state of the docker image
         """
+
+        def image_availability_state(image_spec, success_message, failure_message):
+            """
+            Try to get the image specified by image_spec from local. 
+            Return ImageAvailabilityState.
+            """
+            try:
+                image = self._docker.images.get(image_spec)
+                digests = image.attrs.get('RepoDigests', [image_spec])
+                digest = digests[0] if len(digests) > 0 else None
+                with self._lock:
+                    self._image_cache[digest] = ImageCacheEntry(
+                        id=image.id,
+                        digest=digest,
+                        last_used=time.time(),
+                        virtual_size=image.attrs['VirtualSize'],
+                        marginal_size=image.attrs['Size'],
+                    )
+                return ImageAvailabilityState(
+                    digest=digest, stage=DependencyStage.READY, message=success_message
+                )
+            except Exception:
+                return ImageAvailabilityState(
+                    digest=None, stage=DependencyStage.FAILED, message=failure_message
```

Comment:
Would we instead want to send out the specific failure message for this operation as opposed to a generic failure message ("due to unknown reasons")?

Reply:
Yes, we are sending a specific failure message right now for different cases.